pkg/hsskey/filereader.go

```diff
@@ -156,3 +159,75 @@ func getHssFromFile(verbose bool, verboseDangerous bool, filePaths []string) ([]
 
 	return allHss, nil
 }
+
+// WriteHssToTempFile writes a list of HSS to a tmpfs file where the filepath is returned.
+// See WriteHssToFile for HSS details.
+func WriteHssToTempFile(verbose bool, verboseDangerous bool, hss [][]byte) (string, error) {
+	file, err := os.CreateTemp("", "hss_*")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp file %v", err)
+	}
+
+	err = WriteHssToFile(verbose, verboseDangerous, file, hss)
+	if err != nil {
+		file.Close()
+		os.Remove(file.Name())
+		return "", err
+	}
+
+	file.Close()
+	return file.Name(), nil
+}
+
+// WriteHssToFile writes a list of HSS to an open file.
+// Each HSS key is expected to be 64 bytes long and has a checksum for validation.
+// HSS that fail to validate are not written to the file.
+// Function returns error if no HSS are written or writing can no longer continue.
+func WriteHssToFile(verbose bool, verboseDangerous bool, file *os.File, hss [][]byte) error {
```

Comment:
rather than two verbose bools, with the 4 combinations that apply, would a loglevel make more sense?

Reply:
Oh, fair suggestion. I do rather like having the word dangerous be present instead of reducing it to a number due to the nature of the secrets.
But it's not a hill I'd die on.

I'll propose changing it to this (which makes more sense than original anyways. It would eliminate the unnecessary state while still keeping the verbiage. But if you'd prefer a logLevel I can go with that.
		_if verbose **|| verboseDangerous** {
		       ...
			if verboseDangerous {_

Reply:
up to you, but if you have four combinations of options, you need to test all four.
And if you are determined to produce output, you really ought to produce it to an io.Writer, not just call log.Printf from a package. The io.Writer can be io.Discard if you want to null data. 

But all of us who run tests locally, or peruse test output from the CI, would *really* appreciate it if you don't just directly call log.Printf in a package. I can tell you, from experience, how much trouble that is when you're trying to find a problem.

In fact, rather than bools for verbose and verbosedangers, those can be io.Writers if you are determined to have these functions do printing, and can be set to io.Discard in the default case.

Reply:
Marking resolved. Switched to a io.WriteString from an argument. Added combination testing for writer being nil.